src/features/generic/time-utils.ts

```diff
@@ -56,6 +56,31 @@ namespace TimeUtilities {
       displayInternal: displayInternal,
     }
   }
-};
+
+  export function calculateWorkedTimeAndBillableHours(user: TimeEntryTotalDto | DailyCombinedData){
+
+    // calculates billable hours and adds percentage sign
+    const billableHours = (user.projectTime/user.expected * 100).toFixed(1); 
+    const billableHoursWithPercentage = billableHours.concat("%");
+
+    //turns undertime hours positive and calculates hours and minutes
+    const undertimeHours = Math.floor(user.total * -1 / 60)
+    const undertimeMinutes = user.total * -1 % 60;
+
+    //calculates overtime hours and minutes
+    const overtimeHours = Math.floor(user.total / 60)
+    const overtimeMinutes = user.total % 60;
```

Comment:
There calculations seem a bit redundant in this context since the user can only have less than the expected amount, exactly the expected amount or over the expected amount of work time.

So I would change this so that you compare the excepted amount (from user object) to the user total and determine did user go under, equal or above the expected work time. And based on that you construct single message that can be in similar format as you already have, but the "Undertime"/"Overtime" would be determined from the values.

In addition, use Luxons Duration.fromObject for getting the duration in hours/minutes.

Reply:
Done